libs/design/scss/theming/contrast/contrast-ratio/max-contrast.scss

```diff
@@ -0,0 +1,39 @@
+@use 'sass:list';
+@use './contrast-ratio' as cr;
+
+/// Returns the color from a list of colors that has the highest contrast ratio with a reference color.
+/// If the list is empty, an error is thrown. If the list has one color, that color is returned.
+/// Uses the cr.daff-contrast-ratio function to calculate contrast ratios.
+///
+/// @group Theming
+///
+/// @param {Color} $reference-color
+/// @param {List} $colors — A list of colors to compare against the reference color
+/// @return {Color} The color from the color list that has the highest contrast ratio with the reference color
+///
+@function daff-max-contrast(
```

Comment:
Can you add test coverage to this? We use https://www.oddbird.net/true/docs/

You can see an example test here: https://github.com/graycoreio/daffodil/blob/develop/libs/design/scss/core/map/map-get/map-get.spec.scss